net/route.go

```diff
@@ -29,14 +25,31 @@ func overlaps(n1, n2 *net.IPNet) bool {
 
 // For a specific address, we only care if it is actually *inside* an
 // existing route, because weave-local traffic never hits IP routing.
-func CheckAddressOverlap(addr net.IP) error {
+func CheckAddressOverlap(addr net.IP, ignoreIfNames []string) error {
+	return forEachRoute(ignoreIfNames, func(route netlink.Route) error {
+		if route.Dst != nil && route.Dst.Contains(addr) {
+			return fmt.Errorf("Address %s overlaps with existing route %s on host.", addr, route.Dst)
+		}
+		return nil
+	})
+}
+
+func forEachRoute(ignoreIfNames []string, check func(netlink.Route) error) error {
 	routes, err := netlink.RouteList(nil, netlink.FAMILY_V4)
 	if err != nil {
 		return err
 	}
+routesLoop:
 	for _, route := range routes {
-		if route.Dst != nil && route.Dst.Contains(addr) {
-			return fmt.Errorf("Address %s overlaps with existing route %s on host.", addr, route.Dst)
+		if link, err := netlink.LinkByIndex(route.LinkIndex); err == nil {
+			for _, ignoreIfName := range ignoreIfNames {
+				if link.Attrs().Name == ignoreIfName {
+					continue routesLoop
+				}
+			}
+		}
+		if err := check(route); err != nil {
+			return err
```

Comment:
sort the interface name slice with `sort.Strings(slice)` and then find elements with `sort.SearchStrings(slice, elem)`. Should work out shorter than the code above.